src/ophyd_async/epics/areadetector/writers/general_hdffile.py

```diff
@@ -0,0 +1,129 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Iterator, List, Optional
+from urllib.parse import urlunparse
+
+import event_model
+from event_model import (
+    ComposeStreamResource,
+    ComposeStreamResourceBundle,
+    StreamDatum,
+    StreamResource,
+)
+
+from ophyd_async.core import DirectoryInfo
+
+
+@dataclass
+class _HDFDataset:
+    name: str
```

Comment:
I think this is clearer if we call it:
```suggestion
    #: Name of the data_key within the Descriptor document
    data_key: str
```